src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -48,10 +44,17 @@ FileBasedWal::FileBasedWal(const folly::StringPiece dir,
 
     scanAllWalFiles();
     if (!walFiles_.empty()) {
-        auto& info = walFiles_.rbegin()->second;
         firstLogId_ = walFiles_.begin()->second->firstId();
-        lastLogId_ = info->lastId();
-        lastLogTerm_ = info->lastTerm();
+        auto& info = walFiles_.rbegin()->second;
+        if (info->lastId() <= 0 && ++walFiles_.rbegin() != walFiles_.rend()) {
```

Comment:
maybe check the condition walFiles_.size() >= 2 is faster.

Reply:
haha, you are right 👍 

Don't remember why I chose this ugly condition to check :-(